src/connector/src/source/filesystem/opendal_source/opendal_reader.rs

```diff
@@ -182,3 +223,93 @@ impl<Src: OpendalSource> OpendalReader<Src> {
         }
     }
 }
+
+/// The function `convert_record_batch_to_stream_chunk` is designed to transform the given `RecordBatch` into a `StreamChunk`.
+///
+/// For each column in the source column:
+/// - If the column's schema matches a column in the `RecordBatch` (both the data type and column name are the same),
+///   the corresponding records are converted into a column of the `StreamChunk`.
+/// - If the column's schema does not match, null values are inserted.
+/// - Hidden columns are handled separately by filling in the appropriate fields to ensure the data chunk maintains the correct format.
+/// - If a column in the Parquet file does not exist in the source schema, it is skipped.
+///
+/// # Arguments
+///
+/// * `record_batch` - The `RecordBatch` to be converted into a `StreamChunk`.
+/// * `source_columns` - User defined source schema.
```

Comment:
a little off-topic, but how to parquet files handle schema change

Reply:
The same as json file.